### PR TITLE
Remove code for old gUM flow and use newGumFlow on all clients.

### PR DESCRIPTION
**Pull request by jallamsetty1**
- Remove the code related to old GUM flow and force all clients to use the new GUM flow.
- Force all the clients except React Native to use adapter.js.
- Update the unit test for GUM to use the new gum API.
- Take options.resolution into consideration when constructing the constraints for the GUM call. We were ignoring options.resolution in the new GUM flow APIs.

### CODE REVIEW

File: modules/RTC/JitsiLocalTrack.js [hunk 1]
```diff
@@ -524,30 +514,19 @@ export default class JitsiLocalTrack extends JitsiTrack {
                 facingMode: this.getCameraFacingMode()
             };
 
-            if (browser.usesNewGumFlow()) {
-                promise
-                    = RTCUtils.newObtainAudioAndVideoPermissions(Object.assign(
+            if (this.resolution) {
+                streamOptions.resolution = this.resolution;
+            }
+            promise = RTCUtils.newObtainAudioAndVideoPermissions(Object.assign(
```
Comment (saghul) [inline modules/RTC/JitsiLocalTrack.js]: maybe drop the `new` from the name?

File: modules/RTC/JitsiLocalTrack.js [hunk 2]
```diff
@@ -88,21 +86,13 @@ export default class JitsiLocalTrack extends JitsiTrack {
...
-            this._constraints = track.getConstraints();
-        } else {
-            // FIXME Currently, Firefox is ignoring our constraints about
-            // resolutions so we do not store it, to avoid wrong reporting of
-            // local track resolution.
-            this.resolution = browser.isFirefox() ? null : resolution;
-        }
+        // Get the resolution from the track itself because it cannot be
+        // certain which resolution webrtc has fallen back to using.
+        this.resolution = track.getSettings().height;
```
Comment (saghul) [inline modules/RTC/JitsiLocalTrack.js]: we don't have this on mobile, it throws an error because it's not implemented

File: modules/RTC/JitsiLocalTrack.js [hunk 3]
```diff
@@ -88,21 +86,13 @@ export default class JitsiLocalTrack extends JitsiTrack {
...
 
+        // Cache the constraints of the track in case of any this track
+        // model needs to call getUserMedia again, such as when unmuting.
+        this._constraints = track.getConstraints();
```
Comment (saghul) [inline modules/RTC/JitsiLocalTrack.js]: ditto

File: modules/RTC/JitsiLocalTrack.js [hunk 4]
```diff
@@ -524,30 +514,19 @@ export default class JitsiLocalTrack extends JitsiTrack {
...
                 // The track kind for presenter track is video as well.
                 const mediaType = this.getType() === MediaType.PRESENTER ? MediaType.VIDEO : this.getType();
-                const streamInfo
-                    = browser.usesNewGumFlow()
-                        ? streamsInfo.find(
-                            info => info.track.kind === mediaType)
-                        : streamsInfo.find(
-                            info => info.mediaType === mediaType);
+                const streamInfo = streamsInfo.find(
+                    info => info.track.kind === mediaType);
```
Comment (saghul) [inline modules/RTC/JitsiLocalTrack.js]: doesn't this fit in a single line now?

File: modules/RTC/RTC.js [hunk 5]
```diff
@@ -260,18 +229,8 @@ export default class RTC extends Listenable {
...
-            : RTCUtils.obtainAudioAndVideoPermissions(options);
-
-        return obtainMediaPromise.then(tracksInfo => {
-            if (usesNewGumFlow) {
-                return _newCreateLocalTracks(tracksInfo);
-            }
-
-            return createLocalTracks(tracksInfo, options);
-        });
+        return RTCUtils.newObtainAudioAndVideoPermissions(options)
```
Comment (saghul) [inline modules/RTC/RTC.js]: let's drop the "new" prefix from the APIs now

File: modules/RTC/RTCUtils.js [hunk 6]
```diff
@@ -13,12 +13,10 @@ import { getLogger } from 'jitsi-meet-logger';
 import GlobalOnErrorHandler from '../util/GlobalOnErrorHandler';
 import JitsiTrackError from '../../JitsiTrackError';
 import Listenable from '../util/Listenable';
-import * as MediaType from '../../service/RTC/MediaType';
 import Resolutions from '../../service/RTC/Resolutions';
 import browser from '../browser';
 import RTCEvents from '../../service/RTC/RTCEvents';
 import screenObtainer from './ScreenObtainer';
-import SDPUtil from '../xmpp/SDPUtil';
```
Comment (saghul) [inline modules/RTC/RTCUtils.js]: sweet!

File: modules/RTC/RTCUtils.js [hunk 7]
```diff
@@ -935,24 +539,11 @@ class RTCUtils extends Listenable {
     * on failure.
     **/
     getUserMediaWithConstraints(um, options = {}) {
-        const constraints = getConstraints(um, options);
+        const constraints = newGetConstraints(um, options);
```
Comment (saghul) [inline modules/RTC/RTCUtils.js]: drop new

File: modules/RTC/RTCUtils.js [hunk 8]
```diff
@@ -1338,8 +794,11 @@ class RTCUtils extends Listenable {
             const audioTracks = avStream.getAudioTracks();
 
             if (audioTracks.length) {
-                const audioStream = new MediaStream(audioTracks);
+                const audioStream = new MediaStream();
 
+                for (let i = 0; i < audioTracks.length; i++) {
```
Comment (saghul) [inline modules/RTC/RTCUtils.js]: why is this change needed? is that media stream constructor not supported on some browsers? if so, please leave a comment.

File: modules/RTC/RTCUtils.js [hunk 9]
```diff
@@ -1350,8 +809,11 @@ class RTCUtils extends Listenable {
             const videoTracks = avStream.getVideoTracks();
 
             if (videoTracks.length) {
-                const videoStream = new MediaStream(videoTracks);
+                const videoStream = new MediaStream();
```
Comment (saghul) [inline modules/RTC/RTCUtils.js]: ditto

File: modules/browser/BrowserCapabilities.js [hunk 10]
```diff
@@ -104,10 +104,21 @@ export default class BrowserCapabilities extends BrowserDetection {
      * @returns {boolean} true if the browser is supported, false otherwise.
      */
     isSupported() {
-        return this.isChromiumBased()
-            || this.isFirefox()
-            || this.isReactNative()
-            || this.isSafariWithWebrtc();
+        const REQUIRED_CHROME_VERSION = 69;
+
+        if (this.isChrome()) {
```
Comment (saghul) [inline modules/browser/BrowserCapabilities.js]: why do we check for Chrome excplicitly, if it's already part of the  this.isChromiumBased() check?

File: modules/browser/BrowserCapabilities.js [hunk 11]
```diff
@@ -276,40 +287,14 @@ export default class BrowserCapabilities extends BrowserDetection {
...
-
     /**
      * Checks if the browser uses webrtc-adapter. All browsers using the new
      * getUserMedia flow and Edge.
      *
      * @returns {boolean}
      */
     usesAdapter() {
-        return this.usesNewGumFlow() || this.isEdge();
+        return this.isSupported() || this.isEdge();
```
Comment (saghul) [inline modules/browser/BrowserCapabilities.js]: This is not correct. The adapter doesn't really support RN, so we shouldn't use it there. Also, I think we can now drop the Edge check, since it's not supported. Edgium is, but that's already covered by the previous check. I think this should read `return !this.isReactNative();`

File: modules/RTC/RTCUtils.js [hunk 12]
```diff
@@ -112,228 +97,6 @@ function initEnumerateDevicesWithCallback() {
...
-                }
-
-                // Old style.
-                constraints.audio.optional.push({
-                    sourceId: options.micDeviceId
-                });
-            }
-
-            // if it is good enough for hangouts...
-            constraints.audio.optional.push(
```
Comment (saghul) [inline modules/RTC/RTCUtils.js]: where are these set now?

File: modules/RTC/RTCUtils.js [hunk 13]
```diff
@@ -112,228 +97,6 @@ function initEnumerateDevicesWithCallback() {
...
-        }
-
-        setResolutionConstraints(
-            constraints, isNewStyleConstraintsSupported, options.resolution);
-    }
-    if (um.indexOf('audio') >= 0) {
-        if (browser.isReactNative()) {
-            // The react-native-webrtc project that we're currently using
-            // expects the audio constraint to be a boolean.
-            constraints.audio = true;
```
Comment (saghul) [inline modules/RTC/RTCUtils.js]: note to self, list this limitation.

File: modules/RTC/RTCUtils.js [hunk 14]
```diff
@@ -353,6 +116,7 @@ function getConstraints(um, options = {}) {
  * @param {Object} options.frameRate - used only for dekstop sharing.
  * @param {Object} options.frameRate.min - Minimum fps
  * @param {Object} options.frameRate.max - Maximum fps
+ * @param {string} options.resolution - resolution constraints.
```
Comment (saghul) [inline modules/RTC/RTCUtils.js]: Why is this here now? `resolution` was part of the old flow, the new flow uses `constraints`, which is already a parameter, with the video constrainsts.

File: modules/RTC/JitsiLocalTrack.js (hunk 1 above)
Comment (jallamsetty1) [reply modules/RTC/JitsiLocalTrack.js]: Sure, I will drop the "new" prefix from all the APIs.

File: modules/RTC/RTCUtils.js (hunk 14 above)
Comment (jallamsetty1) [reply modules/RTC/RTCUtils.js]: Maybe I am confused, I was told that some client apps still use options.resolution to override the default constraints. If that is not the case anymore, I will remove resolution from the options.

File: modules/browser/BrowserCapabilities.js (hunk 10 above)
Comment (jallamsetty1) [reply modules/browser/BrowserCapabilities.js]: Yep we don't need the chrome check.

File: modules/RTC/RTCUtils.js (hunk 14 above)
Comment (saghul) [reply modules/RTC/RTCUtils.js]: Right now, the resolution option in config.js is only used in Firefox, well and anything not using the new gUM flow. I always disliked that the new `constraints` option ignored the legacy `resolution` option, so there is that.

I think what we need here is a decission, more than anything.

File: modules/RTC/RTCUtils.js (hunk 12 above)
Comment (jallamsetty1) [reply modules/RTC/RTCUtils.js]: These are set in newGetConstraints() now. I didn't change the audio constraints like we discussed before, thought it will be better to make those changes in a separate PR after thorough testing on all browsers.

File: modules/RTC/RTCUtils.js (hunk 14 above)
Comment (jallamsetty1) [reply modules/RTC/RTCUtils.js]: Ok, I will bring it up with tin the standup to see what we want to do here.

File: modules/browser/BrowserCapabilities.js (hunk 11 above)
Comment (jallamsetty1) [reply modules/browser/BrowserCapabilities.js]: Ok, didn't know that adapter doesn't support RN, also I wasn't sure why Edge was here when it is not supported.

File: modules/RTC/RTCUtils.js (hunk 8 above)
Comment (jallamsetty1) [reply modules/RTC/RTCUtils.js]: The media stream constructor doesn't seem to work when we use MediaStreamTrackMock(unit tests) instead of MediaStreamTrack.

File: modules/RTC/JitsiLocalTrack.js (hunk 2 above)
Comment (jallamsetty1) [reply modules/RTC/JitsiLocalTrack.js]: For now on RN I went back to using the resolution passed to the constructor of JitsiLocalTrack and I am passing `{ video: true }` as constraints when unmuting video.

Comment (saghul) [reply modules/RTC/JitsiLocalTrack.js]: That's a step backwards from the previous behavior, the resolution was respeccted. Let me look into implementing those 2 to see if we can avoid leaving this for later.

File: modules/RTC/RTCUtils.js (hunk 8 above)
Comment (saghul) [reply modules/RTC/RTCUtils.js]: Aha. Maybe leave a comment? I wouldn't have guessed :-)

File: modules/RTC/JitsiLocalTrack.js (hunk 2 above)
Comment (jallamsetty1) [reply modules/RTC/JitsiLocalTrack.js]: Ok sure, will wait for you to implement it. We have another option as well - we could construct the constraints from the options and then use those constraints while unmuting video (just saying).

File: modules/RTC/RTCUtils.js (hunk 8 above)
Comment (jallamsetty1) [reply modules/RTC/RTCUtils.js]: Sure, will do.

**Comment by saghul** [comment]
can you PTAL? https://github.com/react-native-webrtc/react-native-webrtc/pull/739 it's currently only getConstraints, I'm working on getSettings now.
